static/js/src/advantage/subscribe/react/PurchaseModal.jsx

```diff
@@ -199,34 +88,10 @@ const PurchaseModal = () => {
     });
   };
 
-  const onPayClick = () => {
-    checkoutEvent(GAFriendlyProduct, 3);
-    purchaseMutation.mutate("", {
-      onSuccess: (data) => {
-        //start polling
-        setPendingPurchaseID(data);
-      },
-      onError: (error) => {
-        if (error.message.includes("can only make one purchase at a time")) {
-          setError(
-            <>
-              You already have a pending purchase. Please go to{" "}
-              <a href="/account/payment-methods">payment methods</a> to retry.
-            </>
-          );
-        } else {
-          Sentry.captureException(error);
-          setError(
-            <>
-              Sorry, there was an unknown error with with the payment. Check the
-              details and try again. Contact{" "}
-              <a href="https://ubuntu.com/contact-us">Canonical sales</a> if the
-              problem persists.
-            </>
-          );
-        }
-      },
-    });
+  const closeModal = () => {
+    // I have no idea what's going on here, but without it the modal won't close for guest users. ¯\_(ツ)_/¯
+    buttonRef.current.click();
+    buttonRef.current.click();
```

Comment:
We should fix this before merge. I'll try and look into this as well to get a better idea what's going on here.

Reply:
![image](https://user-images.githubusercontent.com/7452681/133068773-c0257c9f-0d73-4bc5-855d-054e6c18aa10.png)
@ClementChaumel  Looks like you have a duplicate `Cancel` button (each step has a ModalFooter + you define an extra one in `PurchaseModal`.

Reply:
Please check if this solves the issue 🙏 https://github.com/ClementChaumel/ubuntu.com/pull/1

Reply:
Yeah, so the clickHandler is set by the vanilla JS form. and I found that it only managed to attach to a button declared on PurchaseModal.jsx. And not any children so what I'm doing here is to create an invisible cancel button on PurchaseModal and then create a function that clicks this button whenever any of the other two (from ModalFooter) is clicked.

But it's proved flaky at best. I think I'm going to switch to just adding the closeModal function to the window object instead.
Again, not ideal. The best thing to do would be to just incorporate the form in the react app. But then again we already use the window object for a bunch of stuff so I figure it's not a horrible solution? What do you think?